### PR TITLE
DES-2182: Questionnaire asset type

### PR DESCRIPTION
**Pull request by duckonomy**
## Overview: ##
Add RAPP questionnaire file (`.rq`) as an importable asset type.

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [DES-2182](https://jira.tacc.utexas.edu/browse/DES-2182)
* [DES-2380](https://jira.tacc.utexas.edu/browse/DES-2380) (partially)

## Summary of Changes: ##
Added questionnaire asset type (panel icon, allowed assets).
Added questionnaire viewer.

## Testing Steps: ##
1. Run related geoapi https://github.com/TACC-Cloud/hazmapper/pull/118 
2. 1. Import `.rq` file from [test project](https://www.designsafe-ci.org/data/browser/projects/8649738474506818026-242ac116-0001-012/) 
3. Ensure that questionnaire assets display properly in asset panel
4. Ensure that questionnaire viewer renders properly.

## Notes: ##
Concerns:
1. The questionnaire viewer code provided by RAPID team was tightly coupled with the render code, which uses jquery. So I had jquery installed for now. But am still tryiing to look for options to isolate jquery.
3. Used `::ng-host` to properly display the `questionnaireBuilder.css` code (globally accessible css)

### CODE REVIEW

File: geoapi/services/features.py [hunk 1]
```diff
@@ -234,6 +238,47 @@ def fromShapefile(projectId: int, fileObj: IO, metadata: Dict, additional_files:
...
+        lat = data.get('geolocation')[0].get('latitude')
+        point = Point(lng, lat)
+        feat = Feature()
+        feat.project_id = projectId
+        feat.the_geom = from_shape(point, srid=4326)
+        feat.properties = metadata or data
+
+        asset_uuid = uuid.uuid4()
+        base_filepath = make_project_asset_dir(projectId)
+        asset_path = os.path.join(base_filepath, str(asset_uuid) + '.rq')
```
Comment (nathanfranklin) [inline geoapi/services/features.py:263]: should we place the rq (and eventually any assets) in a subfolder of the project?

**Review by nathanfranklin — COMMENTED** [review]
Looks great! 💯 

Should we merge to a feature branch where subsequent questionnaire work be added to?

Also for this PR or a follow on:
* could unit tests (a simple rq fixture) be added?
* consider rely on the file and not add the rq to the properties.

**Comment by nathanfranklin** [comment]
I left some suggestions on our shared channel.

File: geoapi/services/features.py [hunk 2]
```diff
@@ -50,17 +50,21 @@ class FeaturesService:
         'shp',
     )
 
+    RAPP_FILE_EXTENSIONS = (
+        'rq',
+    )
+
     ALLOWED_GEOSPATIAL_FEATURE_ASSET_EXTENSIONS = IMAGE_FILE_EXTENSIONS + VIDEO_FILE_EXTENSIONS
```
[the remark this answers is not in the document]
Comment (nathanfranklin) [reply geoapi/services/features.py:57]: ah, this is if you add a file to an existing feature. so not applicable

File: geoapi/services/features.py (hunk 1 above)
Comment (nathanfranklin) [reply geoapi/services/features.py:262]: I think we should still have this in a subfolder like either one of the following:

- /assets/PROJECTEDID/UUID/UIUD.rq
- /assets/PROJECTEDID/UUID/questionaire.rq

File: conf/nginx.conf [hunk 3]
```diff
@@ -40,7 +40,16 @@ http {
             max_ranges 0;
             expires 30d;
             add_header "Access-Control-Allow-Origin"  *;
+            # Preflighted requests
```
Comment (nathanfranklin) [inline conf/nginx.conf:43]: does this need to be added to `kube/geoapi.yaml`

**Review by nathanfranklin — APPROVED** [review]
Looks good to me 💯 

just left one required change to a unit test and then potentially something to update in the kube config.

**Review by sophia-massie — APPROVED** [review]
LGTM 👍

**Pull request merged**